api/httpclient.go

```diff
@@ -32,10 +32,9 @@ func (s *state) HTTPClient() (*httprequest.Client, error) {
 // RootHTTPClient implements Connection.APICaller.HTTPClient and returns an HTTP
 // client pointing to the API server root path.
 func (s *state) RootHTTPClient() (*httprequest.Client, error) {
-	return s.httpClient(&url.URL{
-		Scheme: s.serverScheme,
-		Host:   s.Addr(),
-	})
+	url := s.Addr()
+	url.Scheme = s.serverScheme
```

Comment:
`s.serverScheme` is only written in one place, and is set to "https".
There's a test set up which does `serverScheme:   params.ServerScheme,` but `params.ServerScheme` is never set to anything.
If we add a const for "https" as mentioned elsewhere, we can drop `s.serverScheme` and simplify things. Thus `RootHTTPClient` and `HTTPClient` will just use the "https" const directly.